modules/gdnative/gdnative.cpp

```diff
@@ -858,12 +868,56 @@ bool GDNativeInstance::get(const StringName &p_name, Variant &r_ret) const {
 		r_ret = *(Variant *)&value;
 		return true;
 	}
+
+	Map<StringName, GDNativeScriptData::Method>::Element *E = script->script_data->methods.find("_get");
+	if (E) {
+		Variant name = p_name;
+		const Variant *args[1] = { &name };
+
+		godot_variant result = E->get().method.method((godot_object *)owner, E->get().method.method_data, userdata, 1, (godot_variant**) args);
+		r_ret = *(Variant *)&result;
+		return true;
+	}
+
 	return false;
 }
 
 void GDNativeInstance::get_property_list(List<PropertyInfo> *p_properties) const {
 	script->get_script_property_list(p_properties);
 	// TODO: dynamic properties
+
+	Map<StringName, GDNativeScriptData::Method>::Element *E =  script->script_data->methods.find("_get_property_list");
+	if (E) {
+		godot_variant result = E->get().method.method((godot_object *)owner, E->get().method.method_data, userdata, 0, NULL);
+		Variant ret = *(Variant *)&result;
+		Array arr = ret;
+
+		for (int i = 0; i < arr.size (); i++) {
+			Dictionary d = arr[i];
+			ERR_CONTINUE(!d.has ("name"))
+			ERR_CONTINUE(!d.has ("type"))
+
+			PropertyInfo pinfo;
+
+			pinfo.type = Variant::Type(d["type"].operator int());
+			ERR_CONTINUE(pinfo.type < 0 || pinfo.type >= Variant::VARIANT_MAX);
+
+			pinfo.name = d["name"];
+			ERR_CONTINUE(pinfo.name == "");
+
+			if (d.has ("hint")) {
+				pinfo.hint = PropertyHint(d["hint"].operator int());
+			}
+			if (d.has ("hint_string")) {
+				pinfo.hint_string = d["hint_string"];
+			}
+			if (d.has ("usage")) {
+				pinfo.usage = d["usage"];
+			}
+
+			p_properties->push_back (pinfo);
+		}
+	}
```

Comment:
@Marqin breaking this into additional functions is pointless, this code is used **once** and *exactly this place* is where it belongs. The `d` could be renamed, yeah. But if you know that `_get_property_list()` returns an array of dictionaries all describing the property then this code is super clear.

This code works and doesn't need to be super extensible. It does the job. If it someday doesn't then I'll rewrite it. In fact, I'm rewriting all of this in a private branch anyway.

Putting stuff in functions just for "cleanness" sake is not the best thing to do, sometimes all you need is some good old sequential code.